crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -3132,8 +3131,29 @@ impl Analyzer<'_, '_> {
     #[cfg_attr(debug_assertions, tracing::instrument(skip_all))]
     pub(crate) fn expand_generics_with_type_args(&mut self, span: Span, ty: Type, type_args: &TypeParamInstantiation) -> VResult<Type> {
         match ty.normalize() {
-            Type::Interface(Interface { type_params, .. })
-            | Type::Alias(Alias { type_params, .. })
+            Type::Interface(Interface { type_params, body, .. }) => {
+                let mut params = HashMap::default();
+                let mut type_params_vec: Vec<TypeParam> = vec![];
+
+                if let Some(type_params) = type_params {
+                    type_params_vec.append(&mut type_params.params.clone());
+                }
+
+                for type_elem in body {
+                    if let TypeElement::Constructor(ConstructorSignature {
+                        type_params: Some(type_params),
+                        ..
+                    }) = type_elem
+                    {
+                        type_params_vec.append(&mut type_params.params.clone());
+                    }
+                }
+                for (param, arg) in type_params_vec.iter().zip(type_args.params.iter()) {
```

Comment:
The type parameters are ignored by this `zip` call

Reply:
I don't know that you told me

After passing the following logic, params have the following values.
```
{
	T#2 : TsAnyKeyword
}
```
at exprs/typeGuards/typeGuardsWithInstanceOfbyconstructorSignature/2.ts

Reply:
I updated it